packages/mongodb-dialects/spring-criteria/src/main/kotlin/com/mongodb/jbplugin/dialects/springcriteria/SpringCriteriaContextExtractor.kt

```diff
@@ -14,13 +16,36 @@ object SpringCriteriaContextExtractor : ConnectionContextExtractor<Project> {
     )
 
     override fun gatherContext(contentRoot: Project): ConnectionContext {
-        val database = extractDatabase(contentRoot)
+        val database = extractDatabaseFromProperties(contentRoot)
+            ?: extractDatabaseFromYaml(contentRoot)
```

Comment:
Could it be that there would be both properties and yaml file? If so how should we define the priority?

Reply:
Spring only allows one and it priorities the property file so it should be safe this way.